examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,300 @@
+            if con_key in system_messages.messages:  # Ensure "con_" exists
+                topics.append((message.topic, key, con_key))
+    return topics
+
+
+def select_debate_topic(system_messages: SystemMessages) -> Optional[tuple]:
+    """Prompt the user to select a debate topic from SystemMessages.
+
+    Dynamically loads debate topics from the SystemMessages object, displays
+    the options to the user, and prompts them to select a topic.
+
+    Args:
+        system_messages (SystemMessages): The object containing debate topics.
+
+    Returns:
+        Optional[tuple]: A tuple containing:
+            - topic_name (str): The selected topic's name.
+            - pro_key (str): The key for the pro side of the debate.
+            - con_key (str): The key for the con side of the debate.
+            Returns None if no topics are available or an error occurs.
+    """
+    topics = extract_topics(system_messages)
+
+    if not topics:
+        logger.error("No topics found in the JSON file.")
+        return None
+
+    # Prepare topic choices for user selection
+    topic_choices = "\n".join(
+        [f"{i + 1}. {topic[0]}" for i, topic in enumerate(topics)]
+    )
+    user_input = Prompt.ask(
+        f"Select a debate topic:\n{topic_choices}",
+        choices=[str(i + 1) for i in range(len(topics))],
+        default="1",
+    )
+    topic_index = int(user_input) - 1
+
+    selected_topic = topics[topic_index]
+    logger.info(f"Selected topic: {selected_topic[0]}")
+    return selected_topic
+
+
+def select_side(topic_name: str) -> Literal["pro", "con"]:
+    """Prompt the user to select their side in the debate.
+
+    Presents the user with a choice to debate on either the pro or con side
+    of the given topic.
+
+    Args:
+        topic_name (str): The name of the debate topic.
+
+    Returns:
+        Literal["pro", "con"]: The selected debate side.
+    """
+    side = Prompt.ask(
+        f"Which side would you like to debate on?\n1. Pro-{topic_name}\n2. Con-{topic_name}",
+        choices=["1", "2"],
+        default="1",
+    )
+    return "pro" if side == "1" else "con"
+
+
+def is_llm_delegate() -> bool:
+    """Prompt the user to decide on LLM delegation.
+
+    Asks the user whether the LLM should autonomously continue the debate
+    without requiring user input.
+
+    Returns:
+        bool: True if the user chooses LLM delegation, False otherwise.
+    """
+    return Confirm.ask(
+        "Would you like the LLM to autonomously continue the debate without "
+        "waiting for user input? or ask for your input first time?",
+        default=False,
+    )
+
+
+def run_debate() -> None:
+    """Execute the main debate logic.
+
+    Orchestrates the debate process, including setup, user input, LLM agent
+    interactions, and final feedback. Handles both user-guided and LLM-
+    delegated debates.
+
+    Raises:
+        Exception: If an unexpected error occurs during the debate process.
+    """
+    try:
```

Comment:
no need to enclose the whole thing in try/except since you're re-raise the exception anyway.